source/extensions/http/header_formatters/preserve_case/preserve_case_formatter.cc

```diff
@@ -0,0 +1,63 @@
+#include "extensions/http/header_formatters/preserve_case/preserve_case_formatter.h"
+
+#include "envoy/extensions/http/header_formatters/preserve_case/v3/preserve_case.pb.h"
+#include "envoy/extensions/http/header_formatters/preserve_case/v3/preserve_case.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace HeaderFormatters {
+namespace PreserveCase {
+
+std::string PreserveCaseHeaderFormatter::format(absl::string_view key) const {
+  const auto remembered_key = original_header_keys_.find(key);
```

Comment:
Maybe include `itr` in the variable name? At first glance I thought we were doing a copy here before copying it out of this function again. RVO might apply so it might not have mattered, so not a big deal

Reply:
seconded